astro/src/content/articles/authentication/bff.mdx

```diff
@@ -0,0 +1,329 @@
+---
+title: Backend For Frontend Authentication | FusionAuth
+description: Learn what the OAuth proxy server is, why it's secure, and how FusionAuth supports it.
+author: FusionAuth
+section: Authentication
+tags: oauth, security, BFF, serverless, server, SPA, React, Angular, Vue, Svelte, cookies, tokens, hosted backend, oauth proxy, token handler pattern
+icon: /img/icons/sso-works-why-care.svg
+darkIcon: /img/icons/sso-works-why-care-dark.svg
+url: https://raw.githubusercontent.com/fusionauth/fusionauth-example-hostedbackend/refs/heads/main
+---
+
+import Aside from 'src/components/Aside.astro';
+import Breadcrumb from 'src/components/Breadcrumb.astro';
+import InlineField from 'src/components/InlineField.astro';
+import InlineUIElement from 'src/components/InlineUIElement.astro';
+import {RemoteCode} from '@fusionauth/astro-components';
+
+
+This article discusses the dangers of storing OAuth tokens in the browser or on a client-side device – a common pattern in serverless architectures like single-page applications (SPAs) and mobile apps – and explains how to upgrade to using an OAuth proxy server. An OAuth proxy is also known as the token handler pattern, a token-mediating backend, or an OAuth backend for frontend (BFF).
```

Comment:
Please remove "token-mediating backend" from the last sentence (upon deeper fact-check, the TMB does actually expose the access token to the front end, which is not happening here). My apologies for my earlier comments being erroneous on this front! 🙇‍♀️